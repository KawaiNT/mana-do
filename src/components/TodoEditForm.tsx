import React, { useRef, useState } from 'react';
import { TodoEditFormInterface } from '../models/todo';
import useOutsideClick from './useOutsideClick'

const TodoEditForm = (props: TodoEditFormInterface) => {
  const wrapperRef = useRef(null)
  const [inputRef, setInputRef] = useState(props.todo?.content)

  const onChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputRef(e.target.value)
  }

  const onKeyDown = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter' && inputRef) {
      props.onUpdateTodo({ ...props.todo, content: inputRef})
    }
  }

  useOutsideClick(wrapperRef, () => {
    props.onCancelUpdate()
  })

  return (
    <div ref={wrapperRef} data-testid="todo-edit-form">
      <input
        value={inputRef}
        className="Todo__input__edit"
        data-testid="input-todo-edit"
        onChange={onChange}
        onKeyDown={onKeyDown}
      />
    </div>
  )
}

export default TodoEditForm