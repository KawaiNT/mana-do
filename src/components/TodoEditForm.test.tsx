import React from 'react';
import { render } from '@testing-library/react';
import TodoEditForm from './TodoEditForm'

describe('TodoEditForm component test', () => {
  test('TodoEditForm render', () => {
    const todo = {
      id: '1',
      user_id: '1',
      content : 'Test',
      created_date: '2021-12-06T15:57:42.254Z',
    }
    const onUpdateTodo = jest.fn()
    const onCancelUpdate = jest.fn()
    const { asFragment } = render(<TodoEditForm todo={todo} onUpdateTodo={onUpdateTodo} onCancelUpdate={onCancelUpdate} />)

    expect(asFragment(<TodoEditForm todo={todo} onUpdateTodo={onUpdateTodo} onCancelUpdate={onCancelUpdate} />)).toMatchSnapshot()
  })

  test('TodoEditForm should be in the document', () => {
    const todo = {
      id: '1',
      user_id: '1',
      content : 'Test',
      created_date: '2021-12-06T15:57:42.254Z',
    }
    const onUpdateTodo = jest.fn()
    const onCancelUpdate = jest.fn()
    const { getByTestId } = render(<TodoEditForm todo={todo} onUpdateTodo={onUpdateTodo} onCancelUpdate={onCancelUpdate} />)
    const todoEditForm = getByTestId('todo-edit-form')

    expect(todoEditForm).toBeInTheDocument()
  })
})