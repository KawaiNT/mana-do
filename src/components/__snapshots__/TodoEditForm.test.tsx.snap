// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoEditForm component test TodoEditForm render 1`] = `
<DocumentFragment>
  <div
    data-testid="todo-edit-form"
  >
    <input
      class="Todo__input__edit"
      data-testid="input-todo-edit"
      value="Test"
    />
  </div>
</DocumentFragment>
`;
